File: src/main/java/com/sudosoo/takeiteasy/service/CommentServiceImpl.java
package com.sudosoo.takeiteasy.service;

import com.sudosoo.takeiteasy.common.service.JpaService;
import com.sudosoo.takeiteasy.dto.comment.CreateCommentRequestDto;
import com.sudosoo.takeiteasy.entity.Comment;
import com.sudosoo.takeiteasy.entity.Post;
import com.sudosoo.takeiteasy.repository.CommentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional
public class CommentServiceImpl implements CommentService , JpaService<Comment,Long> {
    private final CommentRepository commentRepository;
    private final PostService postService;

    @Override
    public JpaRepository<Comment, Long> getJpaRepository() {
        return commentRepository;
    }

    @Override
    public Comment create(CreateCommentRequestDto createCommentRequestDto){
        //TODO MemberSetting
        Long memberId = createCommentRequestDto.getMemberId();
        Post post = postService.getByPostId(createCommentRequestDto.getPostId());
        Comment comment = Comment.of(createCommentRequestDto);
        comment.setPost(post);
        comment.setMember(memberId);

        return saveModel(comment);
    }

    @Override
    public Comment getByCommentId(Long commentId) {
        return findModelById(commentId);
    }

    @Override
    public Page<Comment> getCommentsByPostId(Long postId, Pageable pageRequest) {
        return commentRepository.findCommentsByPostId(postId,pageRequest);
    }


}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/application/service/CommentServiceImpl.kt
package com.sudosoo.takeItEasy.application.service

import com.sudosoo.takeItEasy.application.common.service.JpaService
import com.sudosoo.takeItEasy.application.dto.comment.CreateCommentRequestDto
import com.sudosoo.takeItEasy.domain.entity.Category
import com.sudosoo.takeItEasy.domain.entity.Comment
import com.sudosoo.takeItEasy.domain.entity.Post
import com.sudosoo.takeItEasy.domain.repository.CommentRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class CommentServiceImpl(
    val commentRepository: CommentRepository,
    val postService: PostService
) : CommentService, JpaService<Comment, Long>{
    override var jpaRepository: JpaRepository<Comment, Long> = commentRepository

    override fun create(createCommentRequestDto: CreateCommentRequestDto): Comment {
        //TODO MemberSetting
        val memberId: Long = createCommentRequestDto.memberId
        val post: Post = postService.getByPostId(createCommentRequestDto.postId)
        val comment = Comment(createCommentRequestDto.content)
        comment.post = post
        comment.setMemberId(memberId)

        return save(comment)
    }

    override fun getByCommentId(commentId: Long): Comment {
        return findById(commentId)
    }

    override fun getCommentPaginationByPostId(postId: Long, pageRequest: Pageable): Page<Comment> {
        return commentRepository.findCommentsByPostId(postId, pageRequest)
    }
}
